packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -75,6 +79,46 @@ function markExplicitImport(url: string) {
   return url
 }
 
+async function extractImportedBindings(
+  id: string,
+  source: string,
+  importSpec: ImportSpecifier,
+  importedBindings: Map<string, Set<string>>
+) {
+  let bindings = importedBindings.get(id)
+  if (!bindings) {
+    bindings = new Set<string>()
+    importedBindings.set(id, bindings)
+  }
+
+  const isDynamic = importSpec.d > -1
+  const isMeta = importSpec.d === -2
+  if (isDynamic || isMeta) {
+    // this basically means the module will be impacted by any change in its dep
+    bindings.add('*')
+  }
+
+  const exp = source.slice(importSpec.ss, importSpec.se)
+  const [parsed] = parseStaticImports(exp)
+  if (!parsed) {
+    return
+  }
+  if (parsed.sideEffectOnly) {
+    return
+  }
+  if (parsed.starImport) {
+    bindings.add('*')
+  }
+  if (parsed.defaultImport) {
+    bindings.add('default')
+  }
+  if (parsed.namedImports) {
+    for (const { name } of parsed.namedImports) {
+      bindings.add(name)
+    }
+  }
```

Comment:
I wonder if Vite can/should roll its own parser instead of relying on `parse-static-imports`. @poyoho recently made refactors to emptying comments in strings easily, and the import spec could be simple to parse manually (i hope).

Reply:
We could also check out @pi0 and @danielroe mlly, which includes fast regex-based utils to extract imports: https://github.com/unjs/mlly#importexport-analyzes

Reply:
I have replaced `parse-static-imports` with `mlly`. The [function](https://github.com/unjs/mlly/blob/de60f32a6cd9aaeb7bc77c5ca7515f8dc8c7131c/src/analyze.ts#L71) we need looks like it would be easy enough to extract and inline, if desired. Would that be a good idea?

I have also been looking at [parse-imports](https://github.com/TomerAberbach/parse-imports), which would probably do the job too. It's using a lexing approach, that intuitively seems like it could be faster than regexes, but it is implemented in JS so it might not be the case at all, in practice. Also, it would apparently be more cumbersome to extract just the part we need.

`mlly` seems to strike a better balance for our needs.

I'm not too sure why this change has caused some of the test setups to fail. Any idea?

Reply:
> I wonder if Vite can/should roll its own parser

To be honest, I'm interested in it, but this will be a breakthrough change. Not only `import` expression but also lex `assets` expression, `worker` expression and any more. 

The main advantage is to process these matches in the same state machine, and you can directly ignore comments and strings. There is no need to save the original and clean state.

The disadvantages are also obvious. The readability is not as strong as regexp, and the need for a large area of refactors may cause many regression problems.